tests/unit/Lookup/PropertyDataTypeMatcherTest.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+namespace Wikibase\Lib\Tests\Store;
+
+use PHPUnit_Framework_TestCase;
+use Wikibase\DataModel\Entity\PropertyId;
+use Wikibase\DataModel\Services\Lookup\PropertyDataTypeMatcher;
+
+/**
+ * @covers Wikibase\DataModel\Services\Lookup\PropertyDataTypeMatcher
+ *
+ * @licence GNU GPL v2+
+ * @author Thiemo Mättig
+ */
+class PropertyDataTypeMatcherTest extends PHPUnit_Framework_TestCase {
+
+	/**
+	 * @dataProvider propertyDataTypeProvider
+	 */
+	public function testX( $propertyId, $dataType, $expected ) {
+		$lookup = $this->getMock( 'Wikibase\DataModel\Services\Lookup\PropertyDataTypeLookup' );
+		$lookup->expects( $this->once() )
+			->method( 'getDataTypeIdForProperty' )
+			->with( new PropertyId( 'P1' ) )
+			->will( $this->returnValue( 'equal' ) );
+
+		$instance = new PropertyDataTypeMatcher( $lookup );
+		$this->assertSame( $expected, $instance->isMatchingDataType( $propertyId, $dataType ) );
+	}
+
+	public function propertyDataTypeProvider() {
+		$p1 = new PropertyId( 'P1' );
+
+		return array(
+			array( $p1, 'equal', true ),
+			array( $p1, 'not equal', false ),
+		);
+	}
```

Comment:
Just some thoughts: In general, dataprovides are neat if there is a lot of data but for two test cases I think separate test methods are more appropriate.